resources/js/components/Concept/SourceEdit.vue

```diff
@@ -0,0 +1,173 @@
+      ><i class="fa fa-save"></i> Save</b-button
+    >
+    <b-button @click="toggleEditMode">Cancel</b-button>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    conceptId: {
+      type: Number,
+    },
+    conceptSourceId: {
+      type: Number,
+    },
+    canEditVocabulary: false,
+    sourceIndex: null,
+  },
+  mounted() {
+    this.getConceptSource();
+  },
+  data() {
+    return {
+      citation: null,
+      url: null,
+      foundData: null,
+      editMode: false,
+      note: null,
+      isVocabularyEditor: this.canEditVocabulary === true,
+      index: this.sourceIndex,
+      baseURL: process.env.MIX_APP_URL,
+    };
+  },
+  methods: {
+    getConceptSource: function () {
+      if (!this.conceptSourceId) {
+        return;
+      }
+
+      fetch(`${this.baseURL}/api/concept_sources/` + this.conceptSourceId)
+        .then((data) => data.json())
+        .then((data) => {
+          this.citation = data.citation;
+          this.url = data.url;
+          this.foundData = data.found_data;
+          this.note = data.note;
+        });
+    },
+    saveConceptSource: function () {
+      console.log(
+        `Saving ${this.conceptSourceId} ConceptId: ${this.conceptId}, Index: ${this.index}`,
+      );
+      let vm = this;
+      // Check this with Joseph
+      let currentSource = {
+        concept_id: this.conceptId,
+        citation: this.citation,
+        url: this.url,
+        found_data: this.foundData,
+        note: this.note,
+      };
+      if (this.conceptSourceId) {
+        axios
+          .patch(
+            `${this.baseURL}/api/concept_sources/${this.conceptSourceId}`,
+            currentSource,
+          )
+          .then(function (response) {
+            vm.editMode = false;
+            vm.$emit('saved-source', response.data, vm.index);
+            console.log(response);
+          })
+          .catch(function (error) {
+            console.log(error);
+          });
+      } else {
+        axios
+          .post(`${this.baseURL}/api/concept_sources`, currentSource)
+          .then(function (response) {
+            vm.editMode = false;
+            vm.$emit('saved-source', response.data, vm.index);
+            console.log(response);
+          })
+          .catch(function (error) {
+            console.log(error);
+          });
+      }
+    },
+    deleteSource: function () {
+      console.log(`Deleting Source with id ${this.conceptSourceId}`);
+      var vm = this;
+
+      if (!this.conceptSourceId) {
+        vm.$emit('delete-source');
+        return;
+      }
+
+      axios
+        .delete(`${this.baseURL}/api/concept_sources/${this.conceptSourceId}`)
+        .then(function (response) {
+          console.log('Deleted! ', response);
+          vm.$emit('delete-source');
+        })
+        .catch(function (error) {
+          console.log(error);
+        });
+    },
+    toggleEditMode() {
+      this.$parent.toggleEditMode();
```

Comment:
Generally using $parent is a bad idea and this should be an emit that is handled at the parent level. Not enough to warrant not merging, but it is a form of technical debt as the project complexity grows. It makes this component impossible to use outside of this context.